sharding/indexHashedNodesCoordinator.go

```diff
@@ -623,7 +631,14 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartAction(hdr data.HeaderHandler
 			}
 		}
 	}
+
+	nodesConfig := ihgs.nodesConfig[newEpoch]
 	ihgs.mutNodesConfig.Unlock()
+
+	err = ihgs.shuffledOutHandler.Process(nodesConfig.shardID)
```

Comment:
we might think about calling shuffle out only when the epoch start shard header gets finalized.

Reply:
this happens on the SetProcessed of the metachain in the shard. Aren't the headers finalized when processing them in a different shard?

Reply:
added todo